metadata/src/test/java/org/apache/kafka/metalog/LocalLogManager.java

```diff
@@ -406,7 +406,10 @@ void handleSnapshot(SnapshotReader<ApiMessageAndVersion> reader) {
         }
 
         void handleLeaderChange(long offset, LeaderAndEpoch leader) {
+            // Simulate KRaft implementation by first bumping the epoch before assigning a leader
+            listener.handleLeaderChange(new LeaderAndEpoch(OptionalInt.empty(), leader.epoch()));
             listener.handleLeaderChange(leader);
```

Comment:
This should be guarantee by KRaft. Created this Jira to confirm this: https://issues.apache.org/jira/browse/KAFKA-14301